microprofile/WFLY-19835_microprofile_reactive_messaging_otel_with_kafka.adoc

```diff
@@ -0,0 +1,177 @@
+
+Internally in SmallRye Reactive Messaging, the default value for the tracing integration is that it is always enabled for a channel if not specified with above properties.
+
+This RFE is about allowing users to enable tracing via OpenTelemetry in WildFly, and also to be able to continue to ban it.
+
+
+=== User Stories
+
+Users wish to be able to trace messages sent to/from {connector} via reactive messaging using MicroProfile Telemetry.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/{issue-id}[{issue-id}] - Integrate OpenTelemetry with the MP Reactive Messaging {connector} connector
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19846[WFLY-19846] - Promote MicroProfile Telemetry 2.0 to WildFly Standard
+
+
+=== Affected Projects or Components
+* https://github.com/wildfly/wildfly/ - All integration work is done here
+* https://github.com/smallrye/smallrye-reactive-messaging - contains the implementation of the Reactive Messaging connectors. No work is expected to be needed here, unless bugs are found.
+
+=== Other Interested Projects
+None to my knowledge
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+It must be possible to turn the tracing on. To do this a resource called `subsystem=microprofile-reactive-messaging-smallrye/opentelemetry-tracing=config` is introduced. This adds a capability requirement on the MicroProfile Telemetry subsystem, which in turn requires the OpenTelemetry subsystem. The OpenTelemetry subsystem configures how tracing happens.
+
+This resource contains an attribute called `{connector-lower}-connector`, which allows us to control if and how tracing is enabled. It is an enumeration with the following values:
+
+* `NEVER` - This is the default value of the attribute, and also the value that will be used if the `opentelemetry-tracing=config` child resource is not present in the subsystem. This is the same behaviour that has been available today, i.e. the MicroProfile Config properties mentioned in the link:#overview[Overview] will not have any effect, and tracing will be disabled.
+* `OFF` - By default tracing will be disabled, but a user can use the MicroProfile Config properties from the link:#overview[Overview] to turn on tracing for a connector or individual channel in an application.
+* `ON` - Similar to `OFF` but the default is to enable tracing for all channels, unless overridden for the application with the MicroProfile Config properties from the link:#overview[Overview].
+* `ALWAYS` - Tracing is enabled for all channels, and can not be turned off with the MicroProfile Config properties from the link:#overview[Overview].
+
+The MicroProfile Config properties in the  link:#Overview[Overview], are all booleans, and the values have these meanings:
```

Comment:
@kabir can you add requirements for channels, connectors and the order of resolution fo the final value like in https://github.com/wildfly/wildfly/pull/18320/files#diff-cb0945fc010a25c215e621bb884a50a79cce48b93356cc7679642ca0364cd00cR399-R410, please?